extensions/nasa/src/astronauts.tsx

```diff
@@ -0,0 +1,29 @@
+import { Detail, List } from "@raycast/api";
+import { useFetch } from "@raycast/utils";
+import { ENDPOINTS, HEADERS } from "./constants/prefrences";
+import { AstronautsResponse } from "./types/astronauts";
+import AstronautDetail from "./componenets/astronaut";
+
+export default function ASTRONAUTS() {
+  const { isLoading, data } = useFetch(ENDPOINTS.ASTRONAUTS, {
+    headers: HEADERS,
+    mapResult(result: AstronautsResponse) {
+      return {
+        data: result,
+      };
+    },
+    initialData: [],
+  });
+
+  while (isLoading) {
+    return <Detail isLoading={isLoading} markdown={"Loading Astronauts in Space 🚀"} />;
+  }
+
+  return (
+    <List isShowingDetail>
+      {data.results.map((astronaut) => (
+        <AstronautDetail astronaut={astronaut} key={astronaut.id} />
+      ))}
+    </List>
+  );
+}
```

Comment:
```suggestion
import { List } from "@raycast/api";
import { useFetch } from "@raycast/utils";
import { ENDPOINTS, HEADERS } from "./constants/prefrences";
import { AstronautsResponse } from "./types/astronauts";
import AstronautDetail from "./componenets/astronaut";

export default function ASTRONAUTS() {
  const { isLoading, data } = useFetch(ENDPOINTS.ASTRONAUTS, {
    headers: HEADERS,
    mapResult(result: AstronautsResponse) {
      return {
        data: result,
      };
    },
  });

  return (
    <List isLoading={isLoading} isShowingDetail>
      {data?.results?.map((astronaut) => <AstronautDetail astronaut={astronaut} key={astronaut.id} />)}
    </List>
  );
}
```
Maybe something like this for the commands instead 🙂